From 70353e7000669f9d8ab53053e3d881de336c34b1 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 21 Jul 2022 19:37:26 +0300
Subject: [PATCH] [SW-94864] rom: export boot info apis

Export boot info apis to be re-used in other FW components.

Change-Id: I8c23f765f5d8935b03b65a5565721cb44485338b
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 170885c..776a34a 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -483,7 +483,7 @@
 	regmap_writel(0, mmPSOC_SECURITY_SW_STATUS_7);
 }
 
-uint16_t hl_get_boot_info_ver(void)
+uint16_t hl_get_boot_info_ver_rom(void)
 {
 	return HL_BOOT_INFO_VERSION;
 }
diff --git a/arch/gaudi3/arc/arcboot_rom.txt b/arch/gaudi3/arc/arcboot_rom.txt
index a947d0b..a28314f 100644
--- a/arch/gaudi3/arc/arcboot_rom.txt
+++ b/arch/gaudi3/arc/arcboot_rom.txt
@@ -9,3 +9,7 @@
 arcboot_rom_mem_init
 arcboot_rom_get_dma_dest_type
 arcboot_is_valid_address_range
+boot_info_finalize_no_merge
+boot_info_set_no_merge
+boot_info_get_no_merge
+hl_get_boot_info_ver_rom
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 82fe316..817bf28 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -832,7 +832,7 @@
 	return;
 }
 
-uint16_t hl_get_boot_info_ver(void)
+uint16_t hl_get_boot_info_ver_rom(void)
 {
 	return HL_BOOT_INFO_VERSION;
 }
diff --git a/common/boot_info.c b/common/boot_info.c
index 02d4460..a8a42be 100644
--- a/common/boot_info.c
+++ b/common/boot_info.c
@@ -78,7 +78,7 @@
 		return;
 
 	boot_info.ptrs.magic = HL_BOOT_INFO_MAGIC;
-	boot_info.ptrs.desc_ver = hl_get_boot_info_ver();
+	boot_info.ptrs.desc_ver = hl_get_boot_info_ver_rom();
 	boot_info.ptrs.desc_sz = sizeof(struct hl_boot_info_desc);
 	boot_info.ptrs.ptrs_sz = sizeof(struct hl_boot_info_ptrs);
 	boot_info.ptrs.offsets_arr_off = boot_info.ptrs.ptrs_sz;
diff --git a/common/boot_info.h b/common/boot_info.h
index 1d95da3..ae08370 100644
--- a/common/boot_info.h
+++ b/common/boot_info.h
@@ -23,6 +23,6 @@
 void boot_info_finalize_no_merge(void);
 int boot_info_set_no_merge(uint32_t type, const void *data, uint32_t size);
 int boot_info_get_no_merge(uint32_t type, void *data, uint32_t max_size, uint32_t *copy_size);
-uint16_t hl_get_boot_info_ver(void);
+uint16_t hl_get_boot_info_ver_rom(void);
 
 #endif /* _BOOT_INFO_H_ */
